docs/GuidGuidRecommender.md

```diff
@@ -0,0 +1,144 @@
+
+Using this dataset as the basis for recommendations
+entails the following assumptions:
+
+- The set of add-ons appearing in the coinstallation dataset represent
+    the universe of known add-ons for the purpose of generating recommendations.
+- We are only dealing with add-ons that were coinstalled
+    with at least one other add-on on the list
+    (otherwise they would not appear in the file at all).
+- Profiles with only a single add-on installed do not contribute to the counts.
+
+__TODO:__ How are system add-ons handled? Add a note about this.
+
+In computing relevance scores, we also draw on an auxiliary dataset
+containing an overall ranking for each add-on in our list,
+read from the file `guid_install_ranking.json`.
+This is used to break ties when sorting candidates on relevance score,
+and in pruning out rare add-ons.
+It is currently computed from overall add-on installation counts
+obtained from AMO.
+
+This data is stored as an object/dict mapping add-on GUIDs to rank scores.
+On initializing a [`GuidGuidCoinstallRecommender`](../taar_lite/recommenders/guidguid.py#L10),
+it is supplied as `tie_breaker_dict`.
+It is also used as the `ranking_dict` passed to the [`MinInstallPrune`](../taar_lite/recommenders/treatments.py#L36) treatment.
+
+
+# Generating the scored candidate lists
+
+The TAAR-lite recommender is built on the following underlying principle:
+_add-ons which tend to be installed together are similar or complimentary,
+and the number of coinstallations indicates the strength of this relationship._
+Thus, starting from a given add-on, good recommendations for other add-ons
+are those which are most often coinstalled.
+
+Given an add-on A, a natural choice for a candidate list and relevance scores
+is then just the list of add-ons that are coinstalled with A,
+together with their coinstallation counts.
+However, recommendation quality can be improved by making certain adjustments
+to these preliminary scored lists, such as:
+
+- tempering the influence of highly popular add-ons
+    so that they don't drown out more relevant, less common ones
+- pruning out obscure add-ons
+- better balancing relevance against diversity
+- improving the experience for users clicking through a chain of recommendations
+    on AMO, such as pruning out cyclical recommendations.
+
+Final candidate lists and relevance scores are generated
+by applying a sequence of such transformations,
+referred to as __treatments__,
+to the raw coinstallation data.
+As recommendations are based solely on the treated data,
+we are assuming that it encapsulates all the information
+necessary in generating recommendations.
+Treatments are discussed in more detail in terms of their graph representations
+[here](./GraphRepresentation.md#treatments).
+
+Treatments are implemented as extensions of the [`BaseTreatment`](../taar_lite/recommenders/treatments.py#L11) class,
+which apply a transformation via the `treat()` method.
```

Comment:
I don't think this is necessary.

The change of gear from super high-level to low-level was jarring.